src/main/scala/cromwell/engine/backend/Backend.scala

```diff
@@ -50,6 +50,17 @@ trait Backend {
   def initializeForWorkflow(workflow: WorkflowDescriptor): HostInputs
 
   /**
+   * Return the implementation of all WDL functions (e.g. read_lines(), stdout()) used
+   * by this backend
```

Comment:
Definitely,  I'll explain here and then copy/paste/edit in the source:

engineFunctions is meant for retrieving engine-specific function implementations.  The implementation of `WdlStandardLibraryFunctions` classes should be able to implement all of their functions with information from the context object.  See `setupCallEnvironment` for the behavior of the TaskExecutionContext

setupCallEnvironment should set up as much of the environment as needed in order for the WdlStandardLibraryFunctions to operate.  For example, a user expression of `write_lines(['a', 'b', 'c'])` will need to have a concept of a current working directory or GCS bucket of where to store the resulting file so it could return a `WdlFile` object.  setupCallEnvironment should be deterministic:  the same inputs should always produce the same `TaskExecutionContext`.